coralproject/talk
# Change permalink behavior

## What does this PR do?

Show only the Permalink and its parent comment (if applicable) instead of the entire comment stream. Also add a `View all comments` button that takes you back to the regular view before being able to make a comment. This PR sort of ended up being "implement Permalinks fully".

### Also fixed in this PR:
- `View n more replies` button now appears on a permalink page
- replies now appear when viewing a permalinked top-level comment
- the permalink popup now appears in the correct place for top-level comments, long comments
- `View n more replies` correctly loads more replies and displays them
- if a top-level comment is linked, only that comment will be highlighted, not all its replies as well
- polling now happens on the permalink page

### Not in this PR
- polling for more replies on a permalink page

## How do I test this PR?

- go to a comment stream and get a permalink
- on the permalink page, you should see the comment's parent comment and all its children
- if you click the View all comments button, it'll go back to the regular view.
- when you go back to the regular view, the url of the parent page and the iframe should change.

Also:

- create a permalink from a top-level comment, a reply and a long comment to make sure everything still works
